src/vs/workbench/contrib/chat/common/chatAgents.ts

```diff
@@ -384,6 +400,53 @@ export class ChatAgentService implements IChatAgentService {
 
 		return data.impl.provideFollowups(request, result, history, token);
 	}
+
+	private _chatParticipantDetectionProviders = new Map<number, IChatParticipantDetectionProvider>();
+	registerChatParticipantDetectionProvider(handle: number, provider: IChatParticipantDetectionProvider) {
+		this._chatParticipantDetectionProviders.set(handle, provider);
+		return toDisposable(() => {
+			this._chatParticipantDetectionProviders.delete(handle);
+		});
+	}
+
+	async detectAgentOrCommand(request: IChatAgentRequest, history: IChatAgentHistoryEntry[], options: { location: ChatAgentLocation }, token: CancellationToken): Promise<{ agent: IChatAgentData; command?: IChatAgentCommand } | undefined> {
+		// TODO@joyceerhl should we have a selector to be able to narrow down which provider to use
```

Comment:
I think it's fine to assume that there will only ever be one, this is probably only private API for us.